cellprofiler/modules/combineobjects.py

```diff
@@ -116,15 +116,17 @@ def run(self, workspace):
 
         objects_y = workspace.object_set.get_objects(self.objects_y.value)
 
+        dimensions = objects_x.dimensions
+
         assert (
             objects_x.shape == objects_y.shape
         ), "Objects sets must have the same dimensions"
 
-        labels_x = objects_x.segmented.copy()
-        labels_y = objects_y.segmented.copy()
+        labels_x = objects_x.segmented.copy().astype("uint16")
+        labels_y = objects_y.segmented.copy().astype("uint16")
 
         output = self.combine_arrays(labels_x, labels_y)
-        output_labels = skimage.morphology.label(output)
+        output_labels = skimage.morphology.label(output).astype("uint16")
```

Comment:
Forcing type here shouldn't be necessary, did you find you needed this?

Reply:
You're right, it's not necessary -- the data type output from skimage.morphology.label is int64, so we were actually scaling back (IIUC)